README.md

```diff
@@ -1,277 +1,899 @@
+swap addresses. **If the wrong beacon information is supplied, assets can be locked forever.** For
+this reason, all minting policies check that the datums have the proper information; the mint
+transaction will fail if they do not.
+
+The `Rational` type is a fraction (decimal types do not work on-chain). All prices in Cardano-Swaps
+are relative (similar to limit orders in an order-book exchange). Swaps are always priced in
+askedAsset/offeredAsset. For example, if ADA is being offered for DUST at a price of 1.5
+(converted to 3/2), the contract requires that 3 DUST are deposited for every 2 ADA removed from
+the swap address. Ratios of DUST:ADA >= 3/2 will pass, while ratios < 3/2 will fail. 
+
+When engaging in swaps, it is only necessary that the desired swap ratio is met; **not all assets
+in the UTxO must be swapped.** For example, if there is 100 ADA in a swap address requesting 2:1 for
+DUST, a user may swap 20 ADA, as long as they return 80 ADA and 10 DUST in the same Tx. Since every
+user explicitly defines their desired swap ratios, oracles are not required. The "global" price
+naturally emerges where the local bids and asks meet - just like an order-book. 
+
+**All prices for ADA must be in units of Lovelace.**
+
+##### Creating a One-Way Swap
+
+Swap "writers" must first create a swap by using the `CreateSwap` Redeemer in the following manner:
+
+In order to mint beacons with this redeemer, **all of the following must be true**:
+
+1) The beacons must go to an addres protected by the one-way swap validator script.
+2) The beacons must go to an address using a valid staking credential.
+3) The UTxOs with the beacons must have the proper value:
+    - Exactly two kinds of beacons: pair beacon and offer beacon.
+    - The beacons must correspond to the beacons in the datum.
+    - There must be exactly 1 of each beacon.
+    - No extraneous assets are in the UTxO. ADA is always allowed.
+4) The beacons must be stored with the proper inline `SwapDatum`:
+    - `beaconId` == this policy id.
+    - `pairBeacon` == pair beacon asset name for this swap.
+    - `offerId` == asset_id of the offer asset.
+    - `offerName` == asset_name of the offer asset.
+    - `offerBeacon` == offer beacon asset name for this swap's offer asset.
+    - `askId` == asset_id of the ask asset.
+    - `askName` == asset_name of the ask asset.
+    - `swapPrice` denominator > 0
+    - `swapPrice` > 0
+5) The offer asset and ask assets must be different assets.
+
+Once the beacons are minted to the swap address, the spending script does not allow closing the swap
+*unless* the beacons are being burned. This prevents beacons from being sent to unrelated addresses
+and guarantees the integrity of the off-chain queries.
+
+All swaps must be stored with the proper beacons. Due to this requirement, the absolute minimum
+possible value for a swap UTxO to have is about 2 ADA. This minimum value can be reclaimed upon
```

Comment:
That seems strange to me. What about this sentence:
> All prices for ADA must be in units of Lovelace.

Should "ADA" and "Lovelace" be lowercase?

Reply:
This ADA I didn't suggest lowercasing since the user is quoting an exchange price & therefore it is effectively a trading symbol & proper for it to be in full caps.  But you are right that lovelace is also lowercase... I didn't notice this because of having seen the proper name in initial caps so often in other contexts.